### PR TITLE
Bugfix: Corrected loss computation with chunksize > 0

### PR DESCRIPTION
**Pull request by codiceSpaghetti**
This PR resolves an issue in loss computation when chunksize is greater than zero. The correction ensures that only prompts generated by the model contribute to the mean loss computation.

Previously, the mean was calculated across all positions of the sequence, including those within the prompt. In such cases, the loss was automatically set to 0 due to the **correct** use of `torch.nn.functional.cross_entropy` with `ignore_index=-1.` 

However, the subsequent `torch.cat(loss_chunks).mean()` considered all elements of the sequence to compute the mean, leading to an artificial reduction in the loss. This effect becomes more pronounced with longer prompts, as the denominator of the mean increases.

### CODE REVIEW

**Comment by awaelchli** [comment]
Good catch @codiceSpaghetti 
We should extend the test case with a target tensor that has -1 entries.

**Review by awaelchli — APPROVED** [review]
test updated

**Comment by codiceSpaghetti** [comment]
Happy to have contributed to this great open-source project :)
